### PR TITLE
Fixes #23622 - allow generating certs_tar for Katello app

### PR DESCRIPTION
**Pull request by sean797**
Requires https://github.com/theforeman/puppet-certs/pull/199

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -54,18 +54,23 @@ end
 task :generate_parser_caches => [PARSER_CACHE_DIR] do
   caches = [
     "#{PARSER_CACHE_DIR}/katello.yaml",
+    "#{PARSER_CACHE_DIR}/cert_generate_archive.yaml"
```
Comment (beav) [inline Rakefile]: needs a comma here

**Comment by sean797** [comment]
Thanks @beav updated! Tests won't pass because https://github.com/theforeman/puppet-certs/pull/199 is required anyway :)

File: README.md [hunk 2]
```diff
@@ -42,8 +42,9 @@ foreman-installer --scenario                     "katello"\
                   --foreman-proxy-puppetca       "false"
 
 # Generate certificates for installing Foreman proxy on another system
-foreman-proxy-certs-generate --foreman-proxy-fqdn "myforeman-proxy-content.example.com"\
-                                     --certs-tar    ~/myforeman-proxy-content.example.com-certs.tar
+katello-generate-cert-archive --server-fqdn "myforeman-proxy-content.example.com"\
```
Comment (ehelms) [inline README.md]: Could we drop the katello naming aspect?

File: hooks/boot/02-message-helpers.rb [hunk 3]
```diff
@@ -67,18 +82,68 @@ def proxy_instructions_message(kafo)
...
      parameters, see <%= color("foreman-installer --scenario foreman-proxy-content --help", :info) %> and
      documentation for more info on setting up additional services):
 
+MSG
+    end
+
+    def foreman_app_instructions_message(certs_tar, server_fqdn)
+      candlepin_oauth_key = Kafo::Helpers.read_cache_data("candlepin_oauth_secret")
+      candlepin_host = `hostname -f`.strip
+      say <<MSG
```
Comment (ehelms) [inline hooks/boot/02-message-helpers.rb]: Empty line above here would help with code readability

**Comment by ehelms** [comment]
Is it fair to say the current implementation here aims to move away from the use of foreman-proxy-certs-generate and introduce a new tool but maintain both?

**Comment by sean797** [comment]
> Is it fair to say the current implementation here aims to move away from the use of foreman-proxy-certs-generate and introduce a new tool but maintain both?

yes - assuming everyone is happy with that?

File: README.md (hunk 2 above)
Comment (sean797) [reply README.md]: So just ‘generate-cert-archive’ ?

I think the katello name is important as it’s explicit for what it is for. It’s also consistent with other tools (backup, restore, service, etc... )

Comment (ehelms) [reply README.md]: How about `foreman-generate-cert-archive` ?

**Comment by ehelms** [comment]
Any reason not to simply do a full rename here and now? This is typically a one off script that is not part of automation. Thus the risk to a full rename to me seems minimal compared to carrying technical debt.

**Comment by sean797** [comment]
I'm happy to just rename if we are sure? I use it in [automation](https://github.com/sean797/ansible-role-foreman_installer/blob/master/tasks/katello-proxy-certs.yml#L94) and its used in forklift, so I doubt we are the only ones doing automation with it, hence the deprecation

File: README.md (hunk 2 above)
Comment (sean797) [reply README.md]: Does the Foreman name make sense if the certs archive cannot be used with vanilla foreman? If there is a plan to make it compatiable with Foreman then, I'd be happier naming it that. 

I think the current naming can make the two "brands" (Katello & Foreman) confusing to users, sometimes we swap the names around for no good reason. To me at least the Foreman name implies I would use it with Foreman, but in this case the tool cannot be used with vanilla Foreman.

**Comment by zjhuntin** [comment]
@sean797 @ehelms is there anymore discussion to this PR? I see https://github.com/theforeman/puppet-certs/pull/210 as well, just wondering as it's related to a 3.9 redmine.

**Comment by sean797** [comment]
@zjhuntin since you updated the issue I assume you had a conversation away from here? I'd really like to see the work for split deployments start gaining some traction soon. IMO this issue (the certs aspect) is ready.

**Comment by ekohl** [comment]
This repository has been deprecated and merged into https://github.com/theforeman/foreman-installer (https://github.com/Katello/katello-installer/pull/731). If this is still relevant, please resubmit the PR there.

**Pull request closed without merging**